tests/Midi_gtest.cpp

```diff
@@ -0,0 +1,46 @@
+#include <gtest/gtest.h>
+#include "hid/midi.h"
+
+//get rid of compiler errors over unused args in stubs
+#define UNUSED(x) (void)x
+#define MAX_MESSAGES 3000
+
+using namespace daisy;
+
+class MidiTestTransport{
+    public:
+    MidiTestTransport() {}
+    ~MidiTestTransport() {}
+
+    struct Config {};
+
+    void Init(Config conf) { UNUSED(conf); }
+
+    //stubs to make the handler happy
+    void StartRx() {};
+    size_t Readable() { return 1; }
+    void FlushRx() { }
+    void Tx(uint8_t* buff, size_t size) { UNUSED(buff); UNUSED(size); }
+    uint8_t Rx() { return 1; }    
+    bool RxActive() { return true; }
+
+    private:
+};
+
+TEST(util_MidiHandler, a_stateAfterInit){
+    MidiHandler<MidiTestTransport> midi;
+    MidiHandler<MidiTestTransport>::Config handler_conf;
+    midi.Init(handler_conf);
```

Comment:
I suspect this initialization will be needed for every test. You could consider to make a test fixture that does this for you so you don't have to repeat this over and over

Reply:
Alternatively you could consider parameterized tests, where the test parameters look like this:

```cpp
struct TestParameters
{
    std::string name; // human readable name of the test scenario for displaying in test logs
    std::array<uint8_t> inputStream; // raw input data
    std::array<MidiEvent> expectedEvents; // expected events
};
```

Then you would write a single test that pushes all bytes into the parser and then reads the results one by one, comparing them to the expected results.
You could then provide multiple instantiations of this test for various test cases, e.g. running status, super long sysex, broken/invalid messages, etc.

Not sure how familiar you are with the Googletest framework. It's not hard once you know the basic pattern. I'd be happy to help you out if you like, just let me know.